imports/client/components/FancyEditor.tsx

```diff
@@ -126,19 +128,36 @@ export const MentionSpan = styled.span<{
   display: inline-block;
   overflow-wrap: break-word;
   border-radius: 4px;
-  color: #4649ef;
-  background-color: #ced0ed;
+  color: ${({ theme }) => theme.colors.mentionSpanText};
+  background-color: ${({ theme }) => theme.colors.mentionSpanBackground};
   ${({ $isSelf }) =>
     $isSelf &&
     css`
-      background-color: #4649ef;
-      color: #fff;
+      background-color: ${({ theme }) => theme.colors.mentionSpanBackground};
+      color: ${({ theme }) => theme.colors.mentionSpanText};
     `}
   font-size: 0.8rem;
 `;
 
-const SelectedMentionSpan = styled(MentionSpan)`
-  box-shadow: 0 0 0 2px #b4d5ff;
+export const PuzzleSpan = styled.span<{
+  theme: Theme;
+  $solvedness: Solvedness;
+}>`
+  padding: 2px 3px 3px;
+  margin: 0 1px;
+  vertical-align: baseline;
+  display: inline-block;
+  overflow-wrap: break-word;
+  border-radius: 4px;
+  color: ${({ theme }) => theme.colors.mentionSpanText};
+  background-color: ${({ $solvedness, theme }) => {
+    return theme.colors.solvedness[$solvedness];
+  }};
+  font-size: 0.8rem;
+`;
+
+const SelectedMentionSpan = styled(MentionSpan)<{ theme: Theme }>`
+  box-shadow: 0 0 0 2px ${({ theme }) => theme.colors.selectedMentionSpanShadow};
```

Comment:
This makes use of `selectedMentionSpanShadow`, but not `selectedMentionSpanBackground` or `selectedMentionSpanText`.  If they're not used, remove them from the theme; if they are, let's get that here.